File: tests/test_grader.py
import os
from dragon_runner.src.harness import TournamentHarness
from dragon_runner.src.config import Config
from dragon_runner.src.cli import RunnerArgs

def test_grader_config(config_factory, cli_factory):

    config : Config = config_factory("ConfigGrade.json")
    args : RunnerArgs = cli_factory(**{
        "mode": "tournament",
        "failure_log": "Failures.txt",
        "timeout": 2
    })
    
    harness = TournamentHarness(config=config, cli_args=args) 
    assert harness is not None
    
    harness.run()
    assert os.path.exists(args.failure_log)

File: dragon-runner-rs/tests/test_grader.rs
use std::path::Path;

use dragon_runner_rs::cli::{Mode, RunnerArgs};
use dragon_runner_rs::config::load_config;
use dragon_runner_rs::harness::{TestHarness, TournamentHarness};

fn configs_dir() -> std::path::PathBuf {
    let manifest = Path::new(env!("CARGO_MANIFEST_DIR"));
    manifest.parent().unwrap().join("tests").join("configs")
}

fn config_path(name: &str) -> String {
    configs_dir().join(name).to_string_lossy().into_owned()
}

#[test]
fn test_grader_config() {
    let path = config_path("ConfigGrade.json");
    let config = load_config(&path, None).expect("config should load");

    let failure_log = "Failures_rs.txt";
    // Clean up from previous runs
    let _ = std::fs::remove_file(failure_log);

    let args = RunnerArgs {
        mode: Mode::Tournament,
        failure_log: failure_log.to_string(),
        timeout: 2.0,
        ..Default::default()
    };

    let mut harness = TournamentHarness::new(config, args);
    harness.run();

    assert!(
        Path::new(failure_log).exists(),
        "failure log should have been created"
    );

    // Clean up
    let _ = std::fs::remove_file(failure_log);
}
